api/api.go

```diff
@@ -22,6 +23,10 @@ type Request struct {
 	Mixins []string
 	// The map of parameters to use.
 	Params map[string][]string
+	// The type of request (i.e. GET/POST etc.)
+	ReqType string
```

Comment:
This feels to me like it should be an enumeration rather than a string eg something like (excuse my bad Go syntax):

```go
type ReqType int
const (
    ReqGet ReqType = iota
    ReqPost
    ReqPut
    ReqDelete // ...
)

func (r ReqType) String() string {
    switch r {
        case ReqGet:
            return "GET"
        // ...
    }
}

// ...
    ReqType ReqType // maybe these two shouldn't be named the same
```

Reason is mostly to guard against something incorrect being put in `ReqType`, and maybe marginally speed up comparisons like 'is this `POST`'.

OTOH, this adds a lot of code for only a small amount of added security, so I'd appreciate a second opinion.  One possible middle approach might be to wrap `string` instead, or just declare a load of `string` constants for GET, POST, etc?

Reply:
Implemented as originally suggested, seemed reasonable enough.